urllib3/fields.py

```diff
@@ -36,11 +36,11 @@ def format_header_param(name, value):
         result = '%s="%s"' % (name, value)
         try:
             result.encode('ascii')
-        except UnicodeEncodeError:
+        except (UnicodeEncodeError, UnicodeDecodeError):
             pass
         else:
             return result
-    if not six.PY3:  # Python 2:
+    if not six.PY3 and isinstance(value, unicode):  # Python 2:
```

Comment:
Maybe we should use `six.text_type` instead? (Also flake8 fails on unicode for some reason)